### ISSUE TITLE
[DISCUSSÃO] - Definição de Stack

### ISSUE THREAD

**Issue by allanmessias** [allanmessias/PetMais#1]
Faaaala galerinha do meu canal, e ai beeeeleeeeza????

Criada aqui a issue de discussão para a PILHA DE TECNOLOGIAS que vamos utilizar para o projeto PETMAIS. 

Peço que mandem sugestões ae!!

O @Dhayvison bbzinho tinha sugerido buscarmos stacks cujo deploy seja gratuito (??), é isso? Não sei do que se trata.

Eu sugiro alguma coisa fácil de realizar, tipo um LaraVue da vida, ou um NextJS com typescript. 

LaraVue é top e muito delicinha de testes. O Laravel é muito produtivo e a gente já conhece PHP.

<sub> vamos de java </sub>

**Comment by Dhayvison** [comment]
Sim, o Laravel é uma boa. Acredito que dê pra usar os starters kits dele pra adiantar muita coisa.  
Já usei o [Breeze](https://laravel.com/docs/9.x/starter-kits) e gostei da experiência de desenvolvimento.  

Sobre o framework front, só conheço o [React](https://pt-br.reactjs.org/) e acho que o @abnerisrael também. Então ainda teríamos que aprender Vue quase do zero.

No caso de um framework _back_ Node eu sugiro o [AdonisJs](https://adonisjs.com/) pois o acho que o NextJS se adequa mais a aplicações mais estáticas, como sites e blogs.
Além disso o Adonis (como o Laravel) já tem plugins de manipulação de banco de dados e autenticação definidos. No Next essas coisas são feitas mais "na mão".

Sobre a hospedagem. Gosto de colocar meus projetos de estudo online. Pois acho que agrega mais usar a aplicação _livein_ do que só ver uma demonstração. Até pouco tempo a heroku fornecia hospedagem Node e PHP free. Mas o serviço foi descontinuado. Poderíamos pesquisar alternativas grátis de hospedagem e caso não encontremos, vida que segue.

Então por enquanto opino em usar uma arquitetura API Laravel com Breeze e React.

**Comment by abnerisrael** [comment]
Também sou a favor da proposta de arquitetura  proposta pelo @Dhayvison , devido a possibilidade de reutilização do código React, tanto para web como para mobile.

**Comment by allanmessias** [comment]
Ai pai paraaa

fechou 

[banco de dados vai ser postgres](https://www.postgresql.org/)

**Comment by Dhayvison** [comment]
Tá bom

**Comment by allanmessias** [comment]
Decidimos para stack utilizar o NextJs, por alguns motivos:

1. Facilidade de trabalhar com backend e frontend dentro do mesmo projeto (o next utiliza node e react);
2. Facilidade de subir o projeto em produção (o Next é da Vercel, então o deploy pode ser feito por lá);
3. Rapidez no build do projeto (TurboPack no NextJs 13 é infinitamente mais rápido que o Webpack e bem mais rápido que o Vite);
4. Muitas ferramentas para trabalhar com autenticação, autorização e sessão de usuários;
5. Curva de aprendizado rápida, por ser Javascript;
6. Elimina a necessidade inicial de containerizar o projeto, visto que Laravel há essa "necessidade", pois o Laravel Sail sobe o projeto em container e adiciona mais complexidade de infraestrutura no projeto;
7. Para a persistência dos dados, continuamos com o Postgres? Ou passamos para outra estrutura?

Para confecção da API, dentro do Next, fazemos em REST ou GraphQL (Tecnologia mais nova)?

**Comment by Dhayvison** [comment]
Next, Postgree e REST

**Comment by Dhayvison** [comment]
Bora, cuida

**Comment by abnerisrael** [comment]
Proponho o uso da arquitetura MVVM no front, para desacoplamento do framework, modularizar o código e facilitar os testes.

![WhatsApp Image 2022-12-16 at 17 52 27](https://user-images.githubusercontent.com/55406717/208669640-eb38efc4-796d-479a-a1c9-0088e79d4d09.jpeg)


Links de referencia:

https://youtu.be/5_rLFMQmYFg (Parte 1)

https://youtu.be/GLA3DHlVF1s (Parte 2)
